src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsClient/EventPipeProvider.cs

```diff
@@ -69,7 +69,7 @@ internal string GetArgumentString()
             {
                 return "";
             }
-            return string.Join(";", Arguments.Select(a => $"{a.Key}={a.Value}"));
+            return string.Join(";", Arguments.Select(a => $"{a.Key}={a.Value.Replace(";", "\";\"")}"));
```

Comment:
I think you need to also escape `=`.  See: https://github.com/dotnet/runtime/blob/af1090c27b82373d30d1ec9c3cefacf3827f3d8b/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs#L59